test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/AddressRegistrationTests.cs

```diff
@@ -149,15 +149,13 @@ public class AddressRegistrationTests
                 host.Start();
 
                 Assert.Equal(5000, host.GetPort());
-
-                var debugLog = testLogger.Messages.Single(log => log.LogLevel == LogLevel.Debug &&
-                    !log.Message.StartsWith("Hosting start", StringComparison.Ordinal)); // Filter out "Hosting starting" and "Hosting started"
-                Assert.True(string.Equals(debugLog.Message, $"No listening endpoints were configured. Binding to {Constants.DefaultIPEndPoint} by default.", StringComparison.Ordinal));
+                Assert.Equal($"No listening endpoints were configured. Binding to {Constants.DefaultIPEndPoint} by default.",
```

Comment:
All we care about is checking that the message is logged. Doesn't matter if there are other messages in there.

BTW, xUnit has `Assert.Single` which makes this even shorter 😄